spec.html

```diff
@@ -38326,6 +38326,126 @@ <h1>`Promise.all` Resolve Element Functions</h1>
         </emu-clause>
       </emu-clause>
 
+      <emu-clause id="sec-promise.allsettled">
+        <h1>Promise.allSettled ( _iterable_ )</h1>
+        <p>The `allSettled` function returns a promise that is fulfilled with an array of promise state snapshots, but only after all the original promises have settled, i.e. become either fulfilled or rejected. It resolves all elements of the passed iterable to promises as it runs this algorithm.</p>
+        <emu-alg>
+          1. Let _C_ be the *this* value.
+          1. If Type(_C_) is not Object, throw a *TypeError* exception.
+          1. Let _promiseCapability_ be ? NewPromiseCapability(_C_).
+          1. Let _iteratorRecord_ be GetIterator(_iterable_).
+          1. IfAbruptRejectPromise(_iteratorRecord_, _promiseCapability_).
+          1. Let _result_ be PerformPromiseAllSettled(_iteratorRecord_, _C_, _promiseCapability_).
+          1. If _result_ is an abrupt completion, then
+            1. If _iteratorRecord_.[[Done]] is *false*, set _result_ to IteratorClose(_iteratorRecord_, _result_).
+            1. IfAbruptRejectPromise(_result_, _promiseCapability_).
+          1. Return Completion(_result_).
+        </emu-alg>
+        <emu-note>
+          <p>The `allSettled` function requires its *this* value to be a constructor function that supports the parameter conventions of the `Promise` constructor.</p>
+        </emu-note>
+    
+        <emu-clause id="sec-performpromiseallsettled" aoid="PerformPromiseAllSettled">
+          <h1>Runtime Semantics: PerformPromiseAllSettled ( _iteratorRecord_, _constructor_, _resultCapability_ )</h1>
+          <p>When the PerformPromiseAllSettled abstract operation is called with arguments _iteratorRecord_, _constructor_, and _resultCapability_, the following steps are taken:</p>
+          <emu-alg>
+            1. Assert: ! IsConstructor(_constructor_) is *true*.
+            1. Assert: _resultCapability_ is a PromiseCapability Record.
+            1. Let _values_ be a new empty List.
+            1. Let _remainingElementsCount_ be a new Record { [[Value]]: 1 }.
```

Comment:
Perhaps it would be beneficial to add a note explaining why `remainingElementsCount` is a Record rather than just a Number (to pass it by reference rather than by value)?

Reply:
This just matches the precedent set by [`PerformPromiseAll`](https://tc39.es/ecma262/#sec-performpromiseall). I'm not opposed to changing this, but maybe we can do it as a separate PR?

Reply:
I find this approach highly distasteful (altho i have no better suggestions atm), and commented such on the proposal repo, but since it does mirror the existing approach i think pursuing changing both in a separate PR is the proper approach (rather than obstructing this proposal).